lib/controllers/fhir_questionnaire/questionnaire_controller.dart

```diff
@@ -67,18 +63,15 @@ class QuestionnaireController extends GetxController {
     /// afterwards, the new UserResponse will be updated to reflect the selected item
     /// then ResponseBoolean will be selected to true for that item
     /// todo: ignore question types that aren't radio-buttons
-    _model.data.surveys.forEach(
-      (s) => s.questions.forEach(
-        (q) => _responsesController.rxMappedActiveResponses.add(
-          q.code,
-          UserResponse(
-            surveyCode: s.code,
-            questionCode: q.code,
-            answerCode: '',
-            responseType: ResponseBoolean(false),
-          ).obs,
-        ),
-      ),
+    _model.data.survey.surveyItems.forEach(
+      (s) => (s as ItemGroup).surveyItems.forEach(
+            (q) => _responsesController.rxMappedActiveResponses.add(
+              (q as Question).linkId,
+              UserResponse(
+                  questionLinkId: q.linkId,
+                  answers: [AnswerBoolean(false)]).obs,
```

Comment:
What does the initial boolean represent now after the refactor? 
Initially, I had a UserResponse created for every possible question that defaulted to false and would change to true when answered/checked. Is that still the case?

Reply:
Answer: this is the single function that decides if an answer has been given (used in validation). In true/false questions, it also lists the true/false boolean.

Reply:
I think you and I are using AnswerBooleans differently. I was only using an AnswerBoolean to record an answer to a boolean question. It looks to me like you're using AnswerBoolean as a marker to say that answer has been selected. If you wanted to keep that, I could add a boolean field to UserResponse called selected, to keep track of which one has been selected. Also, I was making a list of answers to allow for more than one answer to a question. But (if I understand correctly), you're creating a UserResponse for each possible answer. If that's the case, would it make more sense to have a single answer for each userResponse, and then you could pass a list of UserResponses back to me? I can match up the questions and answers using the linkIds, and I'll know if that question is allowed more than one response.